### DISCUSSION TITLE
REST as external Data

### DISCUSSION THREAD

**Discussion by wgbeckmann** [Budibase/budibase#1385] [Help] [answered]
Hi,
I wanted to try the external data REST integration. But I do not find any documentation about ist.
I tried it by my self, but I do not find a way to use the Parameters in the Query string.

Can you please document it.

Thanks,
Wolf

**Comment by joebudi** [comment]
Hey @wgbeckmann 

These two videos should help with querying external data in Budibase:
https://www.youtube.com/watch?v=NIvMwc78DYc&list=PLEel-MMkFaJlq3eLDP36lykR4Xo3EhJq5&index=2
https://www.youtube.com/watch?v=Nei220VBrBU&list=PLEel-MMkFaJlq3eLDP36lykR4Xo3EhJq5&index=2

Thanks, Joe

**Reply by perezstudio** [reply]
I'm sorry but I've followed the Postres Tutorial shown but for a REST API I just get an "ERROR CODE INPUT ERROR" despite all the fields being filled out and parameters being added to the JSON Body it still returns the error. When I try this in postman or insomnia client it works and returns the API Token.

**Reply by mjashanks** [reply]
Hi there - may have seen you in discord with the same issue... pasting my response in here.

Does your request get sent to your API at all?

I.e. is the input error returned from the api or from Budibase?

And are you able to screenshot your JSON? Even with sensitive stuffed redacted.

I will take a random guess - sometimes what gets me is that I forget to put the parameters inside quotes. e.g.

`{ "name" : "{{ nameParameter }}" }`

**Comment by jlperan** [comment, marked as answer]
Hi!

I´m having the very same issue here. I don´t know how to pass parameters to the REST API.
![budibase](https://user-images.githubusercontent.com/5673043/123109170-4c1c1580-d43b-11eb-8d16-c02e994fcc92.png)

If I manually insert the query in the queryString field everything works fine but I am not able to pass the parameter to the API call.

Could you please help?

Thanks!

**Reply by mjashanks** [reply]
Just realised that `id` does not work as a parameter. It must be used for something else under the hood, and we should disallow usage of it.

Call it anything else - `ID` works for me!

**Reply by jlperan** [reply]
You are right!

I´ve been fighting against this for hours. Now it works like a charm.

Thanks!
